skills_ml/ontologies/onet.py

```diff
@@ -30,64 +31,86 @@
 }
 
 
-def build_onet(onet_cache=None):
-    if not onet_cache:
-        onet_cache = OnetSiteCache()
+class Onet(CompetencyOntology):
+    def __init__(self, onet_cache=None):
+        super().__init__()
+        self.onet_cache = onet_cache
+        self.build_onet()
```

Comment:
Since we're doing all this at construction time, I think we can make a few changes for readability:

1. Do `self.onet_cache = onet_cache or OnetSiteCache()` here in the constructor
2. Rename `build_onet` to `_build` and have it assume that self.onet_cache exists.

Plus, since the method is now modifying self instead of returning a new Ontology, to make sure it's idempotent it should either destroy any competencies/occupations/edges it has before building, or simply refuse to run if self already is populated.